docs/blog/2024-12-10-tailoring-nx-for-your-organization.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Tailoring Nx for your organization
+slug: tailoring-nx-for-your-organization
+authors: ['Philip Fulcher']
+tags: [nx]
+cover_image: /blog/images/2024-12-10/header.avif
+---
+
+# Tailoring Nx for your organization
+
+Monorepos enable consistent use of tooling across your organization's many projects. Nx provides an array of plugins with generators and task inference that simplify tool use and configuration. But these plugins are general-use by design: we can’t foresee your exact use case, but we _do_ provide the materials to tailor Nx for your organization.
+
+[Custom plugins](/extending-nx/intro/getting-started) allow you to provide tools, configurations, and utilities to craft Nx for your organization’s specific needs.
+
+## Getting started with custom plugins
+
+Creating a custom plugin couldn’t be easier. [@nx/plugin](/nx-api/plugin) is our plugin for creating new Nx plugins, and it provides a generator for getting started:
+
+```bash
+nx g plugin nx-plugin --directory=plugins --importPath=@org/nx-plugin
+```
+
+We have an [extensive instructions in our docs for building plugins](/extending-nx/intro/getting-started), or you can jump right to the API for [@nx/plugin](/nx-api/plugin).
+
+## Setting up custom generators
+
+Creating custom generators is the biggest bang-for-your-buck change you can make using custom plugins. The generators distributed with Nx plugins have a wide set of options that allow you to fine-tune their output to meet your needs. But if you have decided on certain rules for your organization, such as a directory structure or project tag schema, it can be easy to forget some of your options when running a generator.
+
+Custom generators are an easy way to enforce how generators from Nx plugins are used. For example:
+
+```bash
+nx generate @nx/react:library --name shared-util --directory libs/shared/util --importPath="@org/shared/util" --tags=type:util,scope:shared
+```
+
+To understand the right options for this generator, the engineer needs to remember:
+
+- the project naming convention based on the type of library and its scope
+- the directory structure based on nested folders for scope
+- the import naming convention
+- tag name convention based on type and scope
+
+This is a lot to remember! Even the most diligent engineer will forget one of these or miss a typo. **A custom generator** can solve this problem. Consider a custom generator for your organization that accepts options like this:
+
+```bash
+nx generate @org/plugin:library --name util --scope shared --type util
```

Comment:
We should re-emphasize at the end of the section that this is the command to run the custom generator you just built. I feel it might get lost here as we say "Consider a custom generator for ..." which sounds "hypothetical" etc.. :)